### ISSUE TITLE
Make lastPressed not so global

### ISSUE THREAD

**Issue by denvaar** [denvaar/snek#7]
`lastPressed` is a global variable because it's used inside `gameLoop` which is run recursively, but it's mutated from the key event process handler. I'm not sure how to make this not global yet.

**Comment by euberdeveloper** [comment]
Would it be fine if I fix the issue by putting everything inside app.ts in a class?

**Comment by denvaar** [comment]
Hi @euberdeveloper ! I don't know if I like the idea of everything in a class, but what about a Singleton pattern? Maybe it could work if we had a singleton for `lastPressed` (example: https://www.dofactory.com/javascript/singleton-design-pattern)

**Comment by mleko** [comment]
I think Singleton itself would be global then.

**Comment by denvaar** [comment]
True, but I think it might make it a little bit better than it is currently. I think doing a Singleton would at least show some intent.

**Comment by denvaar** [comment]
Closure might be useful here.
